### PR TITLE
[Display_02] Update display comm interface implementation class

### LINKED ISSUE #163: [Display_02] Print Results in DisplayCommInterfaceImplementation
**Describe the bug**
Currently `DisplayCommInterfaceImplementation` does not print any data to the user.

**Expected behavior**
It should output all the correct data to the user via the console and plotter.

### PR DESCRIPTION
**Pull request by jcortes18**
Closes #163

### CODE REVIEW

**Comment by jcortes18** [comment]
I think it's not passing the gradle tests because TrackDataPlotter hasn't been merged into main.

File: src/main/java/edu/oakland/production/display02/DisplayCommInterfaceImplementation.java [hunk 1]
```diff
@@ -14,6 +15,8 @@
 public class DisplayCommInterfaceImplementation implements DisplayCommInterface {
 
   private DisplayComm2WayInterface comm2Way;
+  private TrackData data;
+  private TrackDataPlotter plotter;
```
Comment (andrewdimmer) [inline src/main/java/edu/oakland/production/display02/DisplayCommInterfaceImplementation.java]: These don't need to be stored at the class level as they are only used at the method level.
```suggestion
```

File: src/main/java/edu/oakland/production/display02/DisplayCommInterfaceImplementation.java [hunk 2]
```diff
@@ -34,6 +37,17 @@ public DisplayCommInterfaceImplementation(DisplayComm2WayInterface displayComm2W
   * @return the TrackData
   */
   public TrackData receiveRfidRequest(int rfidNum) {
-    return comm2Way.passRfidRequest(rfidNum);
+    this.data = comm2Way.passRfidRequest(rfidNum);
```
Comment (andrewdimmer) [inline src/main/java/edu/oakland/production/display02/DisplayCommInterfaceImplementation.java]: If we remove the above private class attributes, then this gets the type added to it.
```suggestion
    TrackData data = comm2Way.passRfidRequest(rfidNum);
```

File: src/main/java/edu/oakland/production/display02/DisplayCommInterfaceImplementation.java [hunk 3]
```diff
@@ -34,6 +37,17 @@ public DisplayCommInterfaceImplementation(DisplayComm2WayInterface displayComm2W
...
+
+    System.out.println("Track Data for RFID tag number " + rfidNum);
+    System.out.println("Location Data Points: " + data.getLocationDataPoints());
+    System.out.println("Course Direction: " + data.getCourseDirection());
+    System.out.println("Speed: " + data.getSpeed());
+    System.out.println("Status: " + data.getStatusMessage());
```
Comment (andrewdimmer) [inline src/main/java/edu/oakland/production/display02/DisplayCommInterfaceImplementation.java]: This should be moved to the `toString()` method of TrackData itself. I would also recommend printing the status message first, then the speed and course IF the `TrackData` object is valid, then the points at the end.

File: src/main/java/edu/oakland/production/display02/DisplayCommInterfaceImplementation.java [hunk 4]
```diff
@@ -34,6 +37,17 @@ public DisplayCommInterfaceImplementation(DisplayComm2WayInterface displayComm2W
...
+
+    this.plotter = new TrackDataPlotter(data);
```
Comment (andrewdimmer) [inline src/main/java/edu/oakland/production/display02/DisplayCommInterfaceImplementation.java]: If we remove the above private class attributes, then this gets the type added to it.
```suggestion
    TrackDataPlotter plotter = new TrackDataPlotter(data);
```

File: src/main/java/edu/oakland/helper/admin/LocationDataPoint.java [hunk 5]
```diff
@@ -69,6 +69,10 @@ public LocalDateTime getTime() {
     return time;
   }
 
+  public String toString() {
+    return lat + " " + lng + " " + time;
```
Comment (andrewdimmer) [inline src/main/java/edu/oakland/helper/admin/LocationDataPoint.java]: I would see if you can make this more descriptive, may something like:
```suggestion
    return "Location Data at " + time.toString() + ":\n\tLatitude: " + lat + "\n\tLongitude: " + lng;
```

**Review by andrewdimmer — CHANGES_REQUESTED** [review]
Couple of quick changes, then I'd like to take over and add some testing, then you should be good to go!

File: src/main/java/edu/oakland/helper/admin/LocationDataPoint.java [hunk 6]
```diff
@@ -69,6 +69,10 @@ public LocalDateTime getTime() {
     return time;
   }
```
Comment (andrewdimmer) [inline src/main/java/edu/oakland/helper/admin/LocationDataPoint.java]: I would make sure you add the `@Override notation and a Javadoc
```suggestion
  /**
   * Creates a print-friendly string of the LocationDataPoint data.
   *
   * @return The print friendly string of the LocationDataPoint data.
   */
  @Override
```

File: src/main/java/edu/oakland/helper/admin/TrackData.java [hunk 7]
```diff
@@ -123,6 +123,20 @@ public float getSpeed() {
     return points;
   }
 
+  @Override
```
Comment (andrewdimmer) [inline src/main/java/edu/oakland/helper/admin/TrackData.java:131]: I would add the following Javadoc here:
```suggestion
  /**
   * Creates a print-friendly string of the TrackData data.
   *
   * @return The print friendly string of the TrackData data.
   */
  @Override
```

File: src/main/java/edu/oakland/helper/admin/TrackData.java [hunk 8]
```diff
@@ -123,6 +123,20 @@ public float getSpeed() {
...
+  public String toString() {
+    StringBuilder str = new StringBuilder();
+    if (isValid()) {
+      System.out.println("Status: " + getStatusMessage());
```
Comment (andrewdimmer) [inline src/main/java/edu/oakland/helper/admin/TrackData.java]: It should always print the status, even if it is not valid.
```suggestion
    str.append("Track Data Status: " + getStatusMessage() + "\n");
    if (isValid()) {
```

File: src/main/java/edu/oakland/helper/admin/TrackData.java [hunk 9]
```diff
@@ -123,6 +123,20 @@ public float getSpeed() {
...
+      System.out.println("Speed: " + getSpeed());
+      System.out.println("Course Direction: " + getCourseDirection());
```
Comment (andrewdimmer) [inline src/main/java/edu/oakland/helper/admin/TrackData.java]: Make sure these get appended to the output!
```suggestion
      str.append("Speed: " + getSpeed() + "\n");
      str.append("Course Direction: " + getCourseDirection() + "\n");
```

**Review by andrewdimmer — CHANGES_REQUESTED** [review]
Few more minor changes, but once you do this you're all set!

File: src/main/java/edu/oakland/helper/admin/LocationDataPoint.java [hunk 10]
```diff
@@ -68,6 +68,15 @@ public float getLng() {
   public LocalDateTime getTime() {
     return time;
   }
+  /**
+   * Creates a print-friendly string of the LocationDataPoint data.
+   *
+   * @return The print friendly string of the LocationDataPoint data.
+   */
+  @Override
```
[the remark this answers is not in the document]
Comment (andrewdimmer) [reply src/main/java/edu/oakland/helper/admin/LocationDataPoint.java:77]: Oops! My bad... let me fix this one...

**Review by andrewdimmer — APPROVED** [review]
Looks good to me! I'm going to fix the last of the test issues caused by the Plotter, then this is set to merge!

**Review by andrewdimmer — APPROVED** [review]
Now, we're all set to merge if everyone else agrees.

**Review by sydneyhill3901 — APPROVED** [review]

**Pull request merged**